### PR TITLE
Pia 2499 refactor how feedback is sent i os bank sdk capture sdk

### PR DESCRIPTION
**Pull request by vizaknaidavid**

### CODE REVIEW

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thank you @vizaknaidavid! Look good. There are some things that I've noticed:

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift [hunk 1]
```diff
@@ -226,8 +226,6 @@ extension GiniBankNetworkingScreenApiCoordinator {
 
 
                 self.giniBankConfiguration.lineItems = result.lineItems
-
-                documentService.resetToInitialState()
```
Comment (zladzeyka) [inline BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift]: [Clarification] where is ` documentService.resetToInitialState()`called?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/GiniNetworkingScreenAPICoordinator.swift [hunk 2]
```diff
@@ -125,11 +127,8 @@ import GiniBankAPILibrary
                 
                 
                 let result = AnalysisResult(extractions: extractions, lineItems: result.lineItems, images: images, document: document)
-                
-                let documentService = self.documentService
-                
+                                
                 self.resultsDelegate?.giniCaptureAnalysisDidFinishWith(result: result)
-                documentService.resetToInitialState()
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/GiniNetworkingScreenAPICoordinator.swift]: [Clarification] ` documentService.resetToInitialState()` needs to be called after the feedback was sent. Where do we call it?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great progress 🎉👏🏻

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift (hunk 1 above)
Comment (vizaknaidavid) [reply BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift]: As you suggested:
`documentService.resetToInitialState() needs to be moved at the end of your cleanup method.`

File: BankSDK/GiniBankSDKExample/Tests/ExtractionFeedbackIntegrationTest.swift [hunk 3]
```diff
@@ -49,7 +46,7 @@ class ExtractionFeedbackIntegrationTest: XCTestCase {
         // (subsequent steps are in CaptureResultsDelegateForExtractionsFeedbackTest)
         self.getExtractionsFromGiniBankSDK(delegate: CaptureResultsDelegateForExtractionsFeedbackTest(integrationTest: self, expect: expect))
 
-        wait(for: [expect], timeout: 30)
+        wait(for: [expect], timeout: 60)
```
Comment (zladzeyka) [inline BankSDK/GiniBankSDKExample/Tests/ExtractionFeedbackIntegrationTest.swift:49]: Let's keep 30 sec - it should be enough or maybe we have a special reason for 60 sec?

**Review by zladzeyka — APPROVED** [review]
Looks good!😊 Great job!
I only notice that timeout increased up to 60 sec - we can keep 30, it should be enough.

File: BankSDK/GiniBankSDKExample/Tests/ExtractionFeedbackIntegrationTest.swift (hunk 3 above)
Comment (vizaknaidavid) [reply BankSDK/GiniBankSDKExample/Tests/ExtractionFeedbackIntegrationTest.swift:49]: With 30 seconds, locally the success rate was not 100%, with this increase it succeeded all the time. But we still have the problem with the CI, so maybe this is not a valid fix and I have to check the problem elsewhere

**Pull request merged**